services/caching-service.json

```diff
@@ -0,0 +1,357 @@
+                  "maxSurge": 1
+               }
+            },
+            "template": {
+               "metadata": {
+                  "labels": {
+                     "application": "${APPLICATION_NAME}",
+                     "deploymentConfig": "${APPLICATION_NAME}"
+                  },
+                  "name": "${APPLICATION_NAME}"
+               },
+               "spec": {
+                  "containers": [
+                     {
+                        "env": [
+                           {
+                              "name": "SERVICE_NAME",
+                              "value": "${APPLICATION_NAME}"
+                           },
+                           {
+                              "name": "SERVICE_PROFILE",
+                              "value": "caching-service"
+                           },
+                           {
+                              "name": "JGROUPS_PING_PROTOCOL",
+                              "value": "openshift.DNS_PING"
+                           },
+                           {
+                              "name": "OPENSHIFT_DNS_PING_SERVICE_NAME",
+                              "value": "${APPLICATION_NAME}-ping"
+                           },
+                           {
+                              "name": "USERNAME",
+                              "valueFrom": {
+                                 "secretKeyRef" : {
+                                    "name" : "${APPLICATION_NAME}",
+                                    "key" : "application-user"
+                                 }
+                              }
+                           },
+                           {
+                              "name": "PASSWORD",
+                              "valueFrom": {
+                                 "secretKeyRef" : {
+                                    "name" : "${APPLICATION_NAME}",
+                                    "key" : "application-password"
+                                 }
+                              }
+                           }
+                        ],
+                        "image": "${IMAGE}",
+                        "livenessProbe": {
+                           "exec": {
+                              "command": [
+                                 "/opt/datagrid/bin/livenessProbe.sh"
+                              ]
+                           },
+                           "initialDelaySeconds": 15,
+                           "timeoutSeconds": 10,
+                           "periodSeconds": 20,
+                           "successThreshold": 1,
+                           "failureThreshold": 5
+                        },
+                        "readinessProbe": {
+                           "exec": {
+                              "command": [
+                                 "/opt/datagrid/bin/readinessProbe.sh"
+                              ]
+                           },
+                           "initialDelaySeconds": 17,
```

Comment:
Any reason why this is 17 and not 15 or 20?

Reply:
Yes apparently it fixes some issues with stability: https://github.com/jboss-container-images/datagrid-7-image/commit/af374d8a10ae463e24fdbc95f1ce04ec61fac1bf